packages/eslint-config-algolia/rules/typescript.js

```diff
@@ -1,11 +1,58 @@
 // eslint-disable-next-line import/no-commonjs
 module.exports = {
   plugins: ['@typescript-eslint'],
+  settings: {
+    'import/parsers': {
+      '@typescript-eslint/parser': ['.ts', '.tsx'],
+    },
+  },
+
   rules: {
     '@typescript-eslint/adjacent-overload-signatures': ['error'],
     '@typescript-eslint/array-type': ['error', { default: 'array-simple' }],
-    '@typescript-eslint/ban-types': ['error'],
-    '@typescript-eslint/ban-ts-comment': ['off'],
+    '@typescript-eslint/ban-types': [
+      'error',
+      {
+        types: {
+          String: {
+            message: 'Use `string` instead.',
+            fixWith: 'string',
+          },
+          Number: {
+            message: 'Use `number` instead.',
+            fixWith: 'number',
+          },
+          Boolean: {
+            message: 'Use `boolean` instead.',
+            fixWith: 'boolean',
+          },
+          Symbol: {
+            message: 'Use `symbol` instead.',
+            fixWith: 'symbol',
+          },
+          Object: {
+            message:
+              'The `Object` type is mostly the same as `unknown`. You probably want `Record<string, unknown>` instead. See https://github.com/typescript-eslint/typescript-eslint/pull/848',
+            fixWith: 'Record<string, unknown>',
+          },
+          '{}': {
+            message:
+              'The `{}` type is mostly the same as `unknown`. You probably want `Record<string, unknown>` instead.',
+            fixWith: 'Record<string, unknown>',
+          },
+          object: {
+            message:
+              'The `object` type is hard to use. Use `Record<string, unknown>` instead. See: https://github.com/typescript-eslint/typescript-eslint/pull/848',
+            fixWith: 'Record<string, unknown>',
+          },
+          Function: 'Use a specific function type instead, like `() => void`.',
+        },
+      },
+    ],
+    '@typescript-eslint/consistent-type-imports': [
+      'error',
+      { prefer: 'type-imports' },
+    ],
```

Comment:
I'm not sure if this one is needed, what's the advantage of splitting up types & other imports?

Reply:
types import can be safely discarded by compilation without having to compute if they are required at runtime.
https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-8.html

Reply:
sounds good!